### PR TITLE
Add upstream cluster-name label to newClusterCR() function.

### PR DESCRIPTION
**Pull request by anvddriesch**
Towards https://github.com/giantswarm/giantswarm/issues/19683
## Checklist

- [ ] Consider SIG UX feedback.
- [ ] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by marians** [comment]
This seems to be missing the generated YAML. `make generate` I think.

File: pkg/apis/infrastructure/v1alpha3/cluster.go [hunk 1]
```diff
@@ -163,6 +163,7 @@ func newClusterCR(obj *AWSCluster, c ClusterCRsConfig) *apiv1alpha3.Cluster {
 		gsLabels := map[string]string{
 			label.ClusterOperatorVersion: c.ReleaseComponents["cluster-operator"],
 			label.Cluster:                c.ClusterID,
+			apiv1alpha3.ClusterLabelName: c.ClusterID,
```
Comment (tfussell) [inline pkg/apis/infrastructure/v1alpha3/cluster.go:166]: Where are you using this function? I understand if you just want to make this small change here, but I would propose that you could alternatively move this function into the operator or component that calls it to reduce the dependence on apiextensions and CAPI libraries.

**Review by axbarsan — APPROVED** [review]

File: pkg/apis/infrastructure/v1alpha3/cluster.go (hunk 1 above)
Comment (anvddriesch) [reply pkg/apis/infrastructure/v1alpha3/cluster.go:166]: This is used by `kubectl-gs`.
I agree with you that having this dependency might get more annoying in the future if we have more changes like this.

**Comment by anvddriesch** [comment]
Closing in favor of https://github.com/giantswarm/kubectl-gs/pull/547

**Pull request closed without merging**